### PR TITLE
Club functionality: object, servlet, frontend form

### LINKED ISSUE #7: Implement Book Clubs
### Features

User will be able to:

- Create a club 

- Add members (by username/email)

- Set the Book Id

- Set the Announcement (String)

- Push Announcement to email

### PR DESCRIPTION
**Pull request by steven-solar**
This PR creates the basic Club Object and servlet on the backend, adds the associated testing and basic frontend form.

### CODE REVIEW

File: capstone-frontend/src/components/CreateClub.js
Comment (antoniolinhart) [inline capstone-frontend/src/components/CreateClub.js]: What's our standard for these backend links? We probably will have to change them on our local machines every time they're cloned so we don't have a diff in this file every time

File: capstone-frontend/src/components/CreateClub.js
Comment (antoniolinhart) [inline capstone-frontend/src/components/CreateClub.js]: You should avoid the `/` tag in the <br> elements
https://google.github.io/styleguide/htmlcssguide.html#Document_Type

File: capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java
Comment (antoniolinhart) [inline capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java]: Remove commented line

File: capstone-backend/src/test/java/com/google/sps/ClubTest.java
Comment (antoniolinhart) [inline capstone-backend/src/test/java/com/google/sps/ClubTest.java]: I feel like these should have some newlines in in it to break up the chunks of tests, like a line after 187 and 201 to separate your arranging step?

File: capstone-backend/src/test/java/com/google/sps/ClubTest.java
Comment (antoniolinhart) [inline capstone-backend/src/test/java/com/google/sps/ClubTest.java]: Empty comment might not be needed here?

File: capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java
Comment (antoniolinhart) [inline capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java]: We should decide on a standard way to structure imports, I usually like doing alphabetical so the diffs when making changes are minimal, but I'm open to something else

**Review by antoniolinhart — APPROVED** [review]
LGTM.

Not sure if you want to do it in this PR or wait, but it could use some tests for the get and post calls on your servlet. For my feature, I'm working on implementing Mockito right now, so I will let you know if I can find a way to make that easier.

File: capstone-frontend/src/components/CreateClub.js
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/CreateClub.js]: I think we should change them to the production backend link. 

In this scenario, it should be ```"/api/clubs"```

File: capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java
Comment (MauroGuerrero20) [inline capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java]: Change it to ```@WebServlet("/api/clubs")``` so it works for the production build.

File: capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java
Comment (MauroGuerrero20) [inline capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java]: Remove ```response.setHeader``` lines for production build.

**Review by MauroGuerrero20 — CHANGES_REQUESTED** [review]
In order for the code to work properly in deployment, you have to change the ```@WebServlet``` and remove the ```response.setHeader``` lines.  

The only problem with changing from **development** to **production** is that if we ```git fetch origin```; we have to change back all ```@WebServlet``` and  ```response.setHeader``` if we are going to develop the code locally. I think we should look for a solution to this problem.

File: capstone-frontend/src/components/CreateClub.js
Comment (steven-solar) [reply capstone-frontend/src/components/CreateClub.js]: fixed!

File: capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java
Comment (steven-solar) [reply capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java]: done!

File: capstone-backend/src/test/java/com/google/sps/ClubTest.java
Comment (steven-solar) [reply capstone-backend/src/test/java/com/google/sps/ClubTest.java]: good idea, will fix.

File: capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java
Comment (steven-solar) [reply capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java]: Makes sense. I like breaking it up by "type" (eg. my classes separate from api separate from java.util structures), but I'll alphabetize within each type

File: capstone-backend/src/test/java/com/google/sps/ClubTest.java
Comment (steven-solar) [reply capstone-backend/src/test/java/com/google/sps/ClubTest.java]: good catch, removing it

File: capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java
Comment (steven-solar) [reply capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java]: fixed!

File: capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java
Comment (steven-solar) [reply capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java]: fixed!

File: capstone-frontend/src/components/CreateClub.js
Comment (steven-solar) [reply capstone-frontend/src/components/CreateClub.js]: This seems to give me an "Unterminated JSX contexts error"

**Comment by MauroGuerrero20** [comment]
Hi Steven, remember to merge with master often to sync up (```git fetch origin; git merge origin/master```). You branch has missing configuration files from master. 

Also it seems like you can't access the React component from the main App.js.

**Comment by steven-solar** [comment]
Thanks Mauro! Rebased on master. Also intentionally left the component out of App.js to avoid merge conflicts later when we redo the frontend.

**Comment by MauroGuerrero20** [comment]
> Thanks Mauro! Rebased on master. Also intentionally left the component out of App.js to avoid merge conflicts later when we redo the frontend.

Ok, I was able to deploy your code on my own gcp project and seems to be working. I did get this warnings just to let you know: 
Screenshot 2020-06-24 at 4 27 24 PM

I also tried to go to "/api/clubs/" and got a 500 status code error. This is probably because I am not calling it properly. 

On the future, feel free to deploy it as well in your GCP to test everything works as you intended.

**Review by MauroGuerrero20 — APPROVED** [review]
LGTM

Also if this PR depends on https://github.com/STEP-Pod-Capstone-Project/Capstone-Project/pull/20, feel free to wait for that one to be merge 😄

File: capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java
Comment (VivekShankar) [inline capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java:57]: Isn't the `id` the same for all of the returned clubs here (the `id` from `request.getParameter("id")` above)? Is this intentional (on the other hand, doesn't each entity `e` have its own `id` that you can use: `e.getProperty('id')`)?

File: capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java
Comment (VivekShankar) [inline capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java]: Can you update this Javadoc comment?

File: capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java
Comment (VivekShankar) [inline capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java:37]: Looks like you're returning the first five clubs here. Is there a plan for the ClubServlet's `doGet` to do something different in the future, maybe return the clubs the user is in for example?

File: capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java
Comment (VivekShankar) [inline capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java:105]: Nice! I don't think we've used `doPut` before - was there a specific reason you wanted to have support for both `doPost` and `doPut` here?

**Review by VivekShankar — APPROVED** [review]
LGTM - great work here Steven, and nice job reviewing Mauro and Antonio! Just left a few comments to clarify some of the functionality here.

File: capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java
Comment (steven-solar) [reply capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java]: Sure thing! Good catch.

File: capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java
Comment (steven-solar) [reply capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java:37]: We do eventually plan to grab clubs based on either search parameters (search by name, book reading) or by membership (see all clubs you're a member of, owner of, or were invited to). As we were discussing the move to Firebase, I decided to leave this as is for now since it may all have to be changed anyways. I just chose 5 as some arbitrary number to stand in in the meantime.

File: capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java
Comment (steven-solar) [reply capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java:105]: I wanted to allow the owner to change the name of the club, the book the club is reading, the club description etc without making a new club. I thought doPut made the most sense for this.

File: capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java
Comment (steven-solar) [reply capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java:57]: Meant to remove the line getting id from the request, as we were having trouble filtering in Datastore. For now our doGet should grab all clubs. I changed it to grab ID from ```e.getProperty("id"); ```

**Pull request merged**
